scripts/ny-empower-income-limits.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * Uses hardcoded income limits sourced from NYSERDA's page on EmPower+,
+ * generate the corresponding income thresholds in low_income_thresholds.json.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { Database, open } from 'sqlite';
+import sqlite3 from 'sqlite3';
+
+const LOW_INCOME_THRESHOLDS_NAME = 'ny-empower-low';
+const MODERATE_INCOME_THRESHOLDS_NAME = 'ny-empower-moderate';
+const SOURCE_URL =
+  'https://www.nyserda.ny.gov/All-Programs/EmPower-New-York-Program/Eligibility-Guidelines';
+
+async function getCountyFips(
+  db: Database,
+  countyName: string,
+): Promise<string> {
+  return (await db.get(
+    "SELECT county_fips FROM zips WHERE county_name = ? AND state_id = 'NY'",
+    countyName,
+  ))!.county_fips;
+}
+
+/**
+ * Convert an array like [100, 200, 333] to an object like:
+ * { "1": 100, "2": 200, "3": 333 }
+ */
+function convertToObject(thresholds: number[]): Record<string, number> {
+  const result: Record<string, number> = {};
+  for (let i = 0; i < thresholds.length; i++) {
+    result[`${i + 1}`] = thresholds[i];
+  }
+  return result;
+}
+
+async function main() {
+  const db = await open({
+    filename: path.join(__dirname, '../incentives-api.db'),
+    driver: sqlite3.Database,
+  });
+  console.log(db);
+
+  const thresholdsPath = path.join(
+    __dirname,
+    '../data/low_income_thresholds.json',
+  );
+  const thresholdsJson = JSON.parse(fs.readFileSync(thresholdsPath, 'utf-8'));
+  const nyJson = thresholdsJson.NY;
+
+  // Low income thresholds under "NY" key
+  nyJson[LOW_INCOME_THRESHOLDS_NAME] = {
+    type: 'hhsize',
+    source_url: SOURCE_URL,
+    // Copy in existing incentives array, if any
+    incentives: nyJson[LOW_INCOME_THRESHOLDS_NAME]?.incentives ?? [],
+    thresholds: convertToObject(data.get('NY')!),
+  };
+
+  // Moderate income thresholds under the name of each county
+  const moderateThresholds: Record<string, Record<string, number>> = {};
+  for (const [countyName, raw] of data.entries()) {
+    if (countyName === 'NY') {
+      continue;
+    }
+
+    const fips = await getCountyFips(db, countyName);
+    console.log(countyName, fips);
+    moderateThresholds[fips] = convertToObject(raw);
+  }
+  console.log(moderateThresholds);
```

Comment:
Similarly, did we mean to leave this log? If so, it could benefit from some additional context ie: `console.log("Generated threshold - ny-empower-moderate:", moderateThresholds);`